javaslang/src/main/java/javaslang/collection/BitSet.java

```diff
@@ -0,0 +1,1024 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _______
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  //  /\__\   JΛVΛSLΛNG
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/ \ /__\ \   Copyright 2014-2016 Javaslang, http://javaslang.io
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/\__\_/  \_//  \__/\_____/   Licensed under the Apache License, Version 2.0
+ */
+package javaslang.collection;
+
+import javaslang.Function1;
+import javaslang.Tuple;
+import javaslang.Tuple2;
+import javaslang.Tuple3;
+import javaslang.control.Option;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.function.*;
+import java.util.stream.Collector;
+
+import static javaslang.collection.Comparators.naturalComparator;
+
+/**
+ * An immutable {@code BitSet} implementation.
+ *
+ * @author Ruslan Sennov
+ * @since 2.1.0
+ */
+public interface BitSet<T> extends SortedSet<T> {
+
+    long serialVersionUID = 1L;
+
+    class Builder<T> {
+
+        final static Builder<Integer> DEFAULT = new Builder<>(i -> i, i -> i);
+
+        final Function1<Integer, T> fromInt;
+        final Function1<T, Integer> toInt;
```

Comment:
If there is no special reason to use `Function1` then please use `Function` instead. In general it is a good practice because it widens the applicability (`Function1` is a `Function`).